components/mime.rst

```diff
@@ -0,0 +1,541 @@
+all the CSS styles**.
+
+CSS inlining means that every HTML tag must define a ``style`` attribute with
+all its CSS styles. This not only increases the email byte size significantly
+but also makes it impossible to manage for complex emails. That's why Twig
+provides a ``CssInlinerExtension`` that automates everything for you. First,
+install the Twig extension in your application:
+
+.. code-block:: terminal
+
+    $ composer require twig/cssinliner-extension
+
+Now, enable the extension (this is done automatically in Symfony applications)::
+
+    // ...
+    use Twig\CssInliner\CssInlinerExtension;
+
+    $templateLoader = new FilesystemLoader(__DIR__.'/templates');
+    $twig = new Environment($templateLoader);
+    $twig->addExtension(new CssInlinerExtension());
+
+Finally, wrap the entire template contents with the ``inline_css`` filter
+(to do so, it's better to use the `{% filter %} Twig tag`_ instead of the
+traditional ``...|filter`` notation):
+
+.. code-block:: html+twig
+
+    {% filter inline_css %}
+        <style>
+            {# here, define your CSS styles as usual #}
+        </style>
+
+        <h1>Welcome {{ username }}!</h1>
+        {# ... #}
+    {% endfilter %}
+
+You can also define some or all CSS style in external files and pass them as
+arguments of the filter:
+
+.. code-block:: html+twig
+
+    {# '@css/' refers to the Twig namespace defined earlier #}
+    {% filter inline_css('@css/mailing.css') %}
+        <style>
+            {# here, define your CSS styles as usual #}
+        </style>
+
+        <h1>Welcome {{ username }}!</h1>
+        {# ... #}
+    {% endfilter %}
+
+Rendering Markdown Contents in Emails with Twig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Twig provides another extension called ``MarkdownExtension`` that lets you
+define the email contents using the `Markdown syntax`_. In addition to the
+extension, you must also install a Markdown conversion library (the extension is
+compatible with all the popular libraries):
+
+.. code-block:: terminal
+
+    $ composer require twig/markdown-extension
+
+    # these libraries are compatible too: erusev/parsedown, michelf/php-markdown
+    $ composer require league/commonmark
+
+Now, enable the extension (this is done automatically in Symfony applications)::
+
+    // ...
+    use Twig\Markdown\MarkdownExtension;
+
+    $templateLoader = new FilesystemLoader(__DIR__.'/templates');
+    $twig = new Environment($templateLoader);
+    $twig->addExtension(new MarkdownExtension());
+
+Finally, use the ``markdown`` filter to convert parts or the entire email
+contents from Markdown to HTML:
+
+.. code-block:: html+twig
+
+    {% filter markdown %}
+        Welcome {{ username }}!
+        =======================
+
+        You signed up to our site using the following email:
+        `{{ email.to }}`
+
+        [Click here to activate your account]({{ url('...') }})
+    {% endfilter %}
+
```

Comment:
You should probably add a note (not here, but that Markdown paragraph makes me think of that) about the fact the generated text part of an email where only the HTML is defined uses Markdown when available, and falls back to `strip_tags`.